src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -632,104 +686,55 @@ private ODataBatchOperationRequestMessage CreateOperationRequestMessageImplement
-        /// Creates an <see cref="ODataBatchOperationResponseMessage"/> for writing an operation of a batch response - implementation of the actual functionality.
-        /// </summary>
-        /// <param name="contentId">The Content-ID value to write in ChangeSet head.</param>
-        /// <returns>The message that can be used to write the response operation.</returns>
-        private ODataBatchOperationResponseMessage CreateOperationResponseMessageImplementation(string contentId)
-        {
-            this.WritePendingMessageData(true);
-
-            // In responses we don't need to use our batch URL resolver, since there are no cross referencing URLs
-            // so use the URL resolver from the batch message instead.
-            this.CurrentOperationResponseMessage = ODataBatchOperationResponseMessage.CreateWriteMessage(
-                this.rawOutputContext.OutputStream,
-                /*operationListener*/ this,
-                this.payloadUriConverter.BatchMessagePayloadUriConverter,
-                this.container);
-            this.SetState(BatchWriterState.OperationCreated);
 
-            Debug.Assert(this.currentOperationContentId == null, "The Content-ID header is only supported in request messages.");
-
-            // write the operation's start boundary string
-            this.WriteStartBoundaryForOperation();
-
-            // write the headers and request separator line
-            ODataBatchWriterUtils.WriteResponsePreamble(this.rawOutputContext.TextWriter, changeSetBoundary != null, contentId);
-
-            return this.CurrentOperationResponseMessage;
-        }
-
-        /// <summary>
-        /// Writes all the pending headers and prepares the writer to write a content of the operation.
-        /// </summary>
-        private void StartBatchOperationContent()
-        {
-            Debug.Assert(this.CurrentOperationMessage != null, "Expected non-null operation message!");
-            Debug.Assert(this.rawOutputContext.TextWriter != null, "Must have a batch writer!");
-
-            // write the pending headers (if any)
-            this.WritePendingMessageData(false);
-
-            // flush the text writer to make sure all buffers of the text writer
-            // are flushed to the underlying async stream
-            this.rawOutputContext.TextWriter.Flush();
+            // reset the size of the current changeset and increase the size of the batch.
+            this.ResetChangeSetSize();
+            this.InterceptException(this.IncreaseBatchSize);
+            this.isInChangset = true;
         }
 
         /// <summary>
-        /// Disposes the batch writer and set the 'OperationStreamRequested' batch writer state;
-        /// called after the flush operation(s) have completed.
+        /// Perform updates after changeset is ended.
         /// </summary>
-        private void DisposeBatchWriterAndSetContentStreamRequestedState()
+        private void UpdateAfterWriteEndChangeset()

```

Comment:
>UpdateAfterWriteEndChangeset [](start = 21, length = 28)

How about "FinishWriteEndChangeset"?

Reply:
sure.

---
In reply to: [155387064](https://github.com/OData/odata.net/pull/992#discussion_r155387064) [](ancestors = 155387064)